src/pages/workspaces/workspace/tools/LaunchAnalysisModal.js

```diff
@@ -58,27 +60,44 @@ export default class LaunchAnalysisModal extends Component {
   }
 
   componentDidMount() {
-    const { workspaceId: { namespace, name }, config: { rootEntityType } } = this.props
+    const { workspaceId: { namespace, name } } = this.props
+    const { entityType } = this.state
 
     Rawls.workspace(namespace, name).entityMetadata().then(
-      entityMetadata => {
-        const { attributeNames, idName } = entityMetadata[rootEntityType]
-        this.setState({ attributeNames, idName })
-      },
+      entityMetadata => this.setState({ entityMetadata, entitySetExists: !!entityMetadata[`${entityType}_set`] }),
       attributeFailure => this.setState({ attributeFailure })
     )
 
-    Rawls.workspace(namespace, name).entitiesOfType(rootEntityType).then(
+    this.loadEntitiesOfType(entityType)
+  }
+
+  loadEntitiesOfType(type) {
+    const { workspaceId: { namespace, name } } = this.props
+
+    Rawls.workspace(namespace, name).entitiesOfType(type).then(
       entities => this.setState({ entities }),
       entityFailure => this.setState({ entityFailure })
     )
   }
 
   renderMain() {
-    const { entities, filterText, launchError, idName, attributeNames, selectedEntity } = this.state
+    const { rootEntityType } = this.props.config
+    const { entityType, entities, entitySetExists, filterText, launchError, entityMetadata, selectedEntity, activeTab } = this.state
+    const { attributeNames, idName } = entityMetadata ? entityMetadata[entityType] : {}
     const filteredEntities = _.filter(entity => entity.name.includes(filterText), entities)
 
     return h(Fragment, [
+      entitySetExists && TabBar({
+        tabs: [{ title: _.capitalize(rootEntityType), key: 'single' }, { title: _.capitalize(rootEntityType + ' Set'), key: 'multiple' }],
+        activeTab,
+        onChangeTab: key => {
+          const newEntityType = `${rootEntityType}${key === 'multiple' ? '_set' : ''}`
+
+          this.setState({ activeTab: key, entities: undefined, entityType: newEntityType })
```

Comment:
Since `entityType` and `activeTab` always change together, we could probably eliminate one and just compute it from the other.